### ISSUE TITLE
2. ASNHC:Mamm Create an Arctos Team

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/data-migration#71]
Follow instructions at [How To Create and Manage Your Arctos Team: Users and Operators](http://handbook.arctosdb.org/how_to/How-to-Create-your-Arctos-Team-Users-and-Operators.html)

**Comment by Jegelewicz** [comment]
@diannakrejsa how are things proceeding? Don't forget that the steps for prepareing to migrate are laid out in your [data migration project](https://github.com/ArctosDB/data-migration/projects/14)!

**Comment by diannakrejsa** [comment]
@Jegelewicz hey! The steps laid out in the data migration project make great sense. There are some per-record issues that I'm trying to straighten out the rules for so the database team and I can make them happen from our Specify extractions. Below are a couple questions that would be helpful to get the final direction on. There is an attachment that may better explain these questions with some example data:

- Confirming we can't import **time stamp** data? Would want to enter a 'Time Stamp Created', 'Time Stamp Modified', and 'Last Modified Agent' if we could, but think you said on the call we can't.

- For fields where there are no values (e.g., records with no reproductive condition and there is a value, unit, date, determiner, etc. series of columns), Arctos will error out if there are data in the other fields but none in the "VALUE" field? Do you sort by a given field with values (ATTRIBUTE_VALUE_3) then copy-paste in date, determiner, etc. for those fields with values? And, determiner is a required field? 

- Do we need an entry on MAX_ERROR_UNITS to submit MAX_ERROR_DISTANCE values? Specify does not have a unit on the extract tables or on the GUI for records with Lat/Long Accuracy, though an accuracy is reported ("750"). I suspect for most the error is in meters or “m” since a lot of the controlled distance language in Specify is in meters, but I can’t be 100% certain. We could either apply “m” as a unit distance for all, or leave all blank with no unit.

Finally are the following data a required part of a Bulkload Specimens file, or can they be a part of some Batch Tools secondary files? 
-  **Preparator**, aka COLLECTOR_AGENT_N 
-  **Preparator Number** aka OTHER_ID_NUM_N 

These are all part of Specify’s Preparations export table, and thereby difficult to merge into the base record file for the initial upload. 

The example data includes "Base Bulkload" as well as a "Part Bulkload" tabs for showing what we're trying to do (with just a few records). Perhaps we can hop on the phone some time so I can better explain my queries and we can quickly go over solutions? Do you have any time in the next couple days for a 20-30min chat once you have a chance to look at the spreadsheet? 

[ASNHC 16320-16335 Bulk & Parts (file for Teresa).xlsx](https://github.com/ArctosDB/data-migration/files/3598699/ASNHC.16320-16335.Bulk.Parts.file.for.Teresa.xlsx)

**Comment by dustymc** [comment]
I'll let Teresa handle the details, but in general...

Modified comes from triggers in Arctos - they're last modified in Arctos. If you do something predictable it should be fairly accessible anyway - [{"modified_by":"whoever","modified_date":"whenever"}] in specimen remarks would be easy to get at, for example.

Fields and dependencies should be described in http://handbook.arctosdb.org/documentation/bulkloader.html. Let us know if anything's not clear and we'll update.

Yes free-text attributes won't accept units - that's just an attempt to not discard data that the user wants.

Units are paired with coordinate error. You could clarify in locality remarks ('error assumed to be in meters' or something should do it).

You can see batch tools here...

<img width="277" alt="Screen Shot 2019-09-10 at 8 55 00 PM" src="https://user-images.githubusercontent.com/5720791/64667059-49d6fb80-d40d-11e9-893d-07353dd432b9.png">

Loading IDs later shouldn't be a problem. Let us know if anything's missing; I can build a tool or SQL it in for you.

Note also that Arctos is case-sensitive - "Unknown" won't find https://arctos.database.museum/agent.cfm?agent_id=0&agent_name=unknown, for example.

**Comment by Jegelewicz** [comment]
> Confirming we can't import time stamp data? Would want to enter a 'Time Stamp Created', 'Time Stamp Modified', and 'Last Modified Agent' if we could, but think you said on the call we can't.

Not as a separate field, but you could place a concatenation of these in the COLLECTION_OBJECT_REMARKS field "Created in Specify: 3/15/2018  2:45:39 PM,Last Modified in Specify: 3/15/2018  2:45:39 PM". This actually might be a good fist issue for you to propose to the community. Adding Attributes of "Original DB Create Date" and "Original DB Last Modified Date" could be useful information to anyone migrating in to Arctos from elsewhere....

> For fields where there are no values (e.g., records with no reproductive condition and there is a value, unit, date, determiner, etc. series of columns), Arctos will error out if there are data in the other fields but none in the "VALUE" field? Do you sort by a given field with values (ATTRIBUTE_VALUE_3) then copy-paste in date, determiner, etc. for those fields with values? And, determiner is a required field?

Yes to all of that. There are more detailed answers on the attached file.

> Do we need an entry on MAX_ERROR_UNITS to submit MAX_ERROR_DISTANCE values? Specify does not have a unit on the extract tables or on the GUI for records with Lat/Long Accuracy, though an accuracy is reported ("750"). I suspect for most the error is in meters or “m” since a lot of the controlled distance language in Specify is in meters, but I can’t be 100% certain. We could either apply “m” as a unit distance for all, or leave all blank with no unit.

Yes units are required with any type of measurement. I agree with Dusty on how to handle the uncertainty.

>>You could clarify in locality remarks ('error assumed to be in meters' or something should do it).

> Finally are the following data a required part of a Bulkload Specimens file, or can they be a part of some Batch Tools secondary files?
> 
>     Preparator, aka COLLECTOR_AGENT_N
>     Preparator Number aka OTHER_ID_NUM_N
> 
> These are all part of Specify’s Preparations export table, and thereby difficult to merge into the base record file for the initial upload.

Not required for initial bulkload. Preparator can be added later using the Bulkload Collector tool: http://arctos.database.museum/tools/BulkloadCollector.cfm and the Preparator numbers using the Bulkload identifiers/relationships tool: http://arctos.database.museum/tools/BulkloadOtherId.cfm

Looking good! Let me know if any of this doesn't make sense!

[ASNHC.16320-16335.Bulk.Parts.file.for.Teresa_response.xlsx](https://github.com/ArctosDB/data-migration/files/3601657/ASNHC.16320-16335.Bulk.Parts.file.for.Teresa_response.xlsx)

**Comment by diannakrejsa** [comment]
Thanks Dusty and Teresa! I'm combining all of these notes as well as conditional changes to convey to the ASU IT team for mass edits. 

Right now I'm establishing a key for specimen part names referencing the Arctos [code table](https://arctos.database.museum/info/ctDocumentation.cfm?table=CTSPECIMEN_PART_NAME) and there are a few missing or ones that seems like they should exist. Shall I submit a separate issue, or is there a reason these particular ones do not exist and they should be named something else?
[Specimen_Part_Name_Additions.xlsx](https://github.com/ArctosDB/data-migration/files/3617094/Specimen_Part_Name_Additions.xlsx)

**Comment by Jegelewicz** [comment]
"heart (lysis   buffer)" add to Mamm Collection (Herp only?) - added to Mamm
"liver (lysis buffer)" add to   Mamm Collection (Herp only?) - added to Mamm
"liver (lysis buffer)" add to   Mamm Collection (Herp only?) - added to Mamm
brain (ethanol) - added to Mamm
cells (ethanol) - suggest use tissue (ethanol) with note in part remark
cells (lysis buffer) - suggest use tissue (lysis buffer) with note in part remark
electrophoretic tissue (frozen) - suggest use tissue (frozen) and place "electrophoretic" in part remarks
heart, kidney (lysis buffer) - added to Mamm
kidney (lysis buffer) - added to Mamm
sperm (ethanol) - added to Mamm
spleen (lysis buffer) - added to Mamm
whole organism, skull-less   (formalin-fixed, 70% ethanol) - suggest use either carcass (formalin-fixed, 70% ethanol) or whole organism (formalin-fixed, 70% ethanol) and put note about skull removal in part remark
wing punch (frozen) - post a separate issue - I assume this is for Bird collections (or is it bats?). If any of the above that were added are needed for other than Mamm collections, let me know. I'm asking for an issue on this one, because it seems like skin should work for this, but perhaps not? @campmlc @jldunnum may have a good answer.

**Comment by diannakrejsa** [comment]
All the above are Mamm collections specimen parts only so far. The wing punch would be from bats, and typically it's taken as non-lethal tissue sampling. I can change those tissues to skin, there is a bit of data implied in it being a wing punch because of the non-vouchered nature of the tissue.

**Comment by campmlc** [comment]
We currently have ear clip (frozen) in addition to skin (frozen). I think wing punch (frozen) is analagous to ear clip (frozen) in that it does indicate a mark release event.  For now I think we should create the wing punch (frozen) part but leave open for further discussion whether these should all be consolidated. I lean to keeping them separate as well because it is becoming common to look at microbiomes and mites etc on different parts of the body surface, so info on where the skin comes from is relevant. 
@jldunnum

**Comment by dustymc** [comment]
PLEASE do not attempt to bury unrelated data in parts. There's a place to explicitly say 'non-lethal.'

The only relevant questions here should be about the parts themselves. If there's something that makes a small piece of wing-skin capable of answering different questions than a small piece or ear-skin can answer then they should be two parts; if not they should be one.

**Comment by Jegelewicz** [comment]
> PLEASE do not attempt to bury unrelated data in parts. There's a place to explicitly say 'non-lethal.'

Absolutely! Make sure that your specimen event type is encounter for these! See [definitions of specimen event types](http://arctos.database.museum/info/ctDocumentation.cfm?table=CTSPECIMEN_EVENT_TYPE)
